Test/domains/urcu-bp.hpp

```diff
@@ -3,22 +3,23 @@
 #include "rcu_domain.hpp"
 
 #include <urcu-bp.h>
-#define _LGPL_SOURCE
 
-namespace std {
-    class rcu_bp: public rcu_domain {
-    public:
-        void register_thread() { rcu_register_thread(); }
-        void unregister_thread() { rcu_unregister_thread(); }
-        void read_lock() noexcept { rcu_read_lock(); }
-        void read_unlock() noexcept { rcu_read_unlock(); }
-        void synchronize() noexcept { synchronize_rcu(); }
-        void retire(rcu_head *rhp, void (*cbf)(rcu_head *rhp)) { call_rcu(rhp, cbf); }
-        void barrier() noexcept { rcu_barrier(); }
-        void quiescent_state() noexcept { rcu_quiescent_state(); }
-        void thread_offline() noexcept { rcu_thread_offline(); }
-        void thread_online() noexcept { rcu_thread_online(); }
+class rcu_domain_bp {
+public:
+    static constexpr bool register_thread_needed() { return false; }
+    void register_thread() { rcu_register_thread(); }
+    void unregister_thread() { rcu_unregister_thread(); }
+    void thread_offline() noexcept { rcu_thread_offline(); }
+    void thread_online() noexcept { rcu_thread_online(); }
 
-        static constexpr bool register_thread_needed() { return false; }
-    };
-} // namespace std
+    static constexpr bool quiescent_state_needed() { return false; }
+    void quiescent_state() noexcept { rcu_quiescent_state(); }
```

Comment:
> what did you have in mind?

Definitely: replacing the body of this function with `{ }` for speed and for clarity (so that nobody else has to ask "why does this function have an effect, if it's explicitly not needed?").

Possibly: eliminating the requirement that RcuDomains implement a `quiescent_state` function at all, if they don't need one. The classically-polymorphic `rcu_domain_wrapper` could automatically fill in a no-op `void quiescent_state() noexcept override { }` for RcuDomains that don't provide such a function.

Possibly x2: eliminating `quiescent_state` from the generic RCU interface altogether. Consider a random user-defined type such as `rcu_domain_qsbr`. It can satisfy the RcuDomain concept *and also* provide a function `quiescent_state` that happens to trigger a cleanup cycle.